diff --git a/build.gradle b/build.gradle
index a5221320..15e99535 100644
--- a/build.gradle
+++ b/build.gradle
@@ -10,6 +10,7 @@ plugins {
 	id 'com.gradleup.shadow' version '9.2.2'
 	id 'com.github.ben-manes.versions' version '0.52.0'
 	id 'org.beryx.runtime' version '2.0.1'
+	id "io.github.jwharm.flatpak-gradle-generator" version "1.8.0"
 }
 
 group = 'net.querz'
@@ -36,6 +37,7 @@ repositories {
 	mavenCentral()
 	maven { url = 'https://jitpack.io/' }
 	maven { url = 'https://plugins.gradle.org/m2/' }
+	maven { url = "./offline-repository" }
 }
 
 dependencies {
@@ -83,6 +85,11 @@ tasks.register('generator', JavaExec) {
 	classpath = sourceSets.main.runtimeClasspath
 }
 
+tasks.flatpakGradleGenerator {
+  outputFile = file("flatpak-sources.json")
+  downloadDirectory = "./offline-repository"
+}
+
 jar {
 	dependsOn minifyCss
 	archiveFileName = "${projectName}-${projectVersion}.jar"
@@ -195,4 +202,4 @@ static minCss(i, o) throws IOException {
 	catch (Exception ex) {
 		ex.printStackTrace()
 	}
-}
\ No newline at end of file
+}
diff --git a/settings.gradle b/settings.gradle
index 6f42e8a9..4436e33b 100644
--- a/settings.gradle
+++ b/settings.gradle
@@ -1 +1,14 @@
+pluginManagement {
+  repositories {
+  	mavenCentral()
+  	maven {
+  		url 'https://jitpack.io/'
+  	}
+  	maven {
+  		url 'https://plugins.gradle.org/m2/'
+  	}
+  	maven { url "./offline-repository" }
+  }
+}
+
 rootProject.name = 'mcaselector'
\ No newline at end of file
-- 
2.55.0

